src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -529,6 +530,45 @@ void optimizeDuplicateOrderByAndDistinct(ASTPtr & query, const Context & context
     DuplicateDistinctVisitor(distinct_data).visit(query);
 }
 
+/// Optimize monotonous functions in ORDER BY
```

Comment:
Useless comment. Let's add some implementation details: "Replace monotonous functions in ORDER BY if they don't participate in GROUP BY expression, has a single argument and not an aggregate functions."